File: app/src/main/java/com/lyf/rrx/rxjava2xdemo/MainActivity.java
package com.lyf.rrx.rxjava2xdemo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import io.reactivex.Observable;
import io.reactivex.ObservableEmitter;
import io.reactivex.ObservableOnSubscribe;
import io.reactivex.functions.Consumer;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Observable.create(new ObservableOnSubscribe<String>() {
            @Override
            public void subscribe(ObservableEmitter<String> emitter) throws Exception {
                emitter.onNext("Hello World!");
            }
        }).subscribe(new Consumer<String>() {
            @Override
            public void accept(String s) throws Exception {
                System.out.println(s);
            }
        });

        Observable.just("Hello World111!").subscribe(new Consumer<String>() {
            @Override
            public void accept(String s) throws Exception {
                System.out.println(s);
            }
        });

        Observable.just("Hello World222!").subscribe(System.out::println);
    }
}

File: chapter17/src/main/java/com/renrenxin/chapter17/MainActivity.kt
package com.renrenxin.chapter17

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val a = "test kotlin"
        val b = 5
        val c: Double = 1.0
        val d: List<String> = ArrayList()

        System.out.println(a.javaClass)
        System.out.println(b.javaClass)
        System.out.println(c.javaClass)
        System.out.println(d.javaClass)
    }
}
